h2o-core/src/main/java/water/fvec/RollupStats.java

```diff
@@ -431,7 +437,9 @@ public void compute2() {
             rs._tsk.join();
           } else if(rs.isMutating()) // c) => throw IAE
             throw new IllegalArgumentException("Can not compute rollup stats while vec is being modified. (3)");
-        } else { // d) => compute the rollups
+        } else { // rs == null d) => compute the rollups
+          final Vec vec = DKV.getGet(_vecKey);
+          if (vec == null) break; // Vec was deleted since we last checked
```

Comment:
Your logic is sound... I would also except an exception if the Vec reference is no longer valid. But none is actually thrown and the problem persists if I remove the check that makes sure the Vec still exists.

I will continue to investigate what is happening. I will also change the code so that exception is thrown if the Vec is deleted (on line 442) instead of just returning (that shouldn't do any harm)